pulsar-client/src/main/java/com/yahoo/pulsar/client/impl/BrokerConsumerStats.java

```diff
@@ -0,0 +1,168 @@
+    private long unackedMessages = 0;
+
+    /** Flag to verify if consumer is blocked due to reaching threshold of unacked messages */
+    private boolean blockedConsumerOnUnackedMsgs = false;
+
+    /** Address of this consumer */
+    private String address = "";
+
+    /** Timestamp of connection */
+    private String connectedSince = "";
+
+    /** Whether this subscription is Exclusive or Shared or Failover */
+    private SubscriptionType subscriptionType = null;
+
+    /** Total rate of messages expired on this subscription. msg/s */
+    private double msgRateExpired = 0;
+
+    /** Number of messages in the subscription backlog */
+    private long msgBacklog = 0;
+
+    public BrokerConsumerStats(CommandConsumerStatsResponse response) {
+        super();
+        this.validTillInMs = System.currentTimeMillis();
+        this.msgRateOut = response.getMsgRateOut();
+        this.msgThroughputOut = response.getMsgThroughputOut();
+        this.msgRateRedeliver = response.getMsgRateRedeliver();
+        this.consumerName = response.getConsumerName();
+        this.availablePermits = response.getAvailablePermits();
+        this.unackedMessages = response.getUnackedMessages();
+        this.blockedConsumerOnUnackedMsgs = response.getBlockedConsumerOnUnackedMsgs();
+        this.address = response.getAddress();
+        this.connectedSince = response.getConnectedSince();
+        this.subscriptionType = SubscriptionType.valueOf(response.getType());
+        this.msgRateExpired = response.getMsgRateExpired();
+        this.msgBacklog = response.getMsgBacklog();
+    }
+
+    public BrokerConsumerStats() {
+    }
+
+    public void setCacheTime(long timeInMs) {
+        validTillInMs = System.currentTimeMillis() + timeInMs;
+    }
+
+    /** Returns true if the Message is Expired **/
+    public synchronized boolean isValid() {
+        return System.currentTimeMillis() <= validTillInMs;
+    }
+
+    public double getMsgRateOut() {
+        return msgRateOut;
+    }
+
+    public double getMsgThroughputOut() {
+        return msgThroughputOut;
+    }
+
+    public double getMsgRateRedeliver() {
+        return msgRateRedeliver;
+    }
+
+    public String getConsumerName() {
+        return consumerName;
+    }
+
+    public long getAvailablePermits() {
+        return availablePermits;
+    }
+
+    public long getUnackedMessages() {
+        return unackedMessages;
+    }
+
+    public boolean isBlockedConsumerOnUnackedMsgs() {
+        return blockedConsumerOnUnackedMsgs;
+    }
+
+    public String getAddress() {
+        return address;
+    }
+
+    public String getConnectedSince() {
+        return connectedSince;
+    }
+
+    public SubscriptionType getSubscriptionType() {
+        return subscriptionType;
+    }
+
+    public double getMsgRateExpired() {
+        return msgRateExpired;
+    }
+
+    public long getMsgBacklog() {
+        return msgBacklog;
+    }
+
+    public synchronized void add(BrokerConsumerStats stats) {
```

Comment:
**just a thought here** for Partitioned consumers: (here we have `add(..)` for to aggregate partitioned-consumers result. But only problem is we can't  get values for individual consumer as all attributes have aggregate value.

```
interface BaseBrokerConsumerStats {
    //getters
    long getAvailablePermits();
    boolean isBlockedConsumerOnUnackedMsgs();
}
```
```
PartitionedBrokerConsumerStats implements BaseBrokerConsumerStats {
    //getters: this will return aggregate result
    long getAvailablePermits();
    boolean isBlockedConsumerOnUnackedMsgs();
    //getters:
    long getAvailablePermits(int partition);
    boolean isBlockedConsumerOnUnackedMsgs(int partition);
}
```
```
BrokerConsumerStats implements BaseBrokerConsumerStats{
    //getters
}
```

and client can cast to `PartitionedBrokerConsumerStats` if they are really interested per partitions else consumerStats will be abstract from client.